client/src/pages/Signup.tsx

```diff
@@ -13,7 +16,21 @@ const Signup = () => {
 
   const submitHandler = (e: FormEvent) => {
     e.preventDefault();
-    setLoading(true);
+
+    if (signup.isLoading) {
+      setLoading(true);
+      return;
+    }
+
+    const userName = usernameRef.current?.value;
+    const name = nameRef.current?.value;
+    const imageUrl = imageRef.current?.value;
+
+    if (!userName || userName === "" || !name || name === "") {
+      return;
```

Comment:
instead of returning render a toast message saying please pass valid input or something :-

https://daisyui.com/components/toast/

also please try using toast once signup API returns 200 ok,

with saying :- `user signup successful, redirecting you to /login route`
and then redirect to /login route